Functions/Engines/Wine/script.js

```diff
@@ -8,333 +8,392 @@ LATEST_STABLE_VERSION = "1.8.6";
  * Wine main prototype
  * @constructor
  */
-var Wine = function () {
-    var that = this;
-    that._wineWebServiceUrl = Bean("propertyReader").getProperty("webservice.wine.url");
-    that._wineEnginesDirectory = Bean("propertyReader").getProperty("application.user.engines.wine");
-    that._winePrefixesDirectory = Bean("propertyReader").getProperty("application.user.wineprefix");
-    that._configFactory = Bean("compatibleConfigFileFormatFactory");
-    that._OperatingSystemFetcher = Bean("operatingSystemFetcher");
-    that._wineDebug = "-all";
-    that._ldPath = Bean("propertyReader").getProperty("application.environment.ld");
-
-    /**
-     *
-     * @param wizard
-     * @returns {Wine}
-     */
-    that.wizard = function (wizard) {
-        that._wizard = wizard;
-        return that;
-    };
+function Wine() {
+    this._wineWebServiceUrl = Bean("propertyReader").getProperty("webservice.wine.url");
+    this._wineEnginesDirectory = Bean("propertyReader").getProperty("application.user.engines.wine");
+    this._winePrefixesDirectory = Bean("propertyReader").getProperty("application.user.wineprefix");
+    this._configFactory = Bean("compatibleConfigFileFormatFactory");
+    this._OperatingSystemFetcher = Bean("operatingSystemFetcher");
+    this._wineDebug = "-all";
+    this._ldPath = Bean("propertyReader").getProperty("application.environment.ld");
+}
 
-    /**
-     *
-     * @param debug
-     * @returns {Wine}
-     */
-    that.debug = function (debug) {
-        if(!debug) {
-            that._wineDebug = "";
-        }
-        that._wineDebug = debug;
-        return that;
-    };
+/**
+*
+* @param {SetupWizard} [wizard]
+* @returns {SetupWizard|Wine}
+*/
+Wine.prototype.wizard = function (wizard) {
+    // get
+    if (arguments.length == 0) {
+        return this._wizard;
+    }
 
-    /**
-     *
-     * @param architecture
-     * @returns {Wine}
-     */
-    that.architecture = function (architecture) {
-        if(arguments.length == 0) {
-            return that._architecture;
-        }
+    // set
+    this._wizard = wizard;
+    return this;
+};
 
-        if(that._prefixConfiguration) {
-            that._prefixConfiguration.writeValue("wineArchitecture", architecture);
-        }
+/**
+*
+* @param {string} [debug]
+* @returns {string|Wine}
+*/
+Wine.prototype.debug = function (debug) {
+    // get
+    if (arguments.length == 0) {
+        this._wineDebug = "";
+        return this._wineDebug;
```

Comment:
This will always return an empty string. What if I want to get the current value?